### ISSUE TITLE
Unexpected behaviour of throws assertion

### ISSUE THREAD

**Issue by vilu** [zio/zio#1608]
```
  import zio.test._
  import zio.test.Assertion._
  val ex = new Exception
  val a = throws(equalTo(ex))
  println(assert(throw ex, a)) // expected Value(Right(())), got exception
```

Maybe I'm missing something here but this seems wrong to me.

**Comment by adamgfraser** [comment]
@vilu Thanks for reporting! Let me take a look and will get back to you.

**Comment by vilu** [comment]
I've checked already, I'm opening up a PR now. Maybe you want to check that first and just merge if you agree?

**Comment by adamgfraser** [comment]
Oh awesome. You're one step ahead. Yes, will do.

**Comment by vilu** [comment]
Done

**Comment by ghostdogpr** [comment]
Closed by #1609
